pkg/webhook/resources/setting/validator.go

```diff
@@ -709,6 +730,201 @@ func (v *settingValidator) validateDeleteStorageNetwork(setting *v1beta1.Setting
 	return werror.NewMethodNotAllowed(fmt.Sprintf("Disallow delete setting name %s", settings.StorageNetworkName))
 }
 
+func (v *settingValidator) getPrometheusVols() ([]string, error) {
```

Comment:
Since for the PVC used by prometheus, its name is determined. So we can directly check PVC cache for its existence. And in case the PVC is manually created with the same name as what prometheus doses, I think we should check prometheus CR first. Thanks.

Reply:
https://github.com/WebberHuang1118/harvester/blob/121f387ad0ce6c83bfde536aeabd02bc6a02663b/pkg/webhook/resources/setting/validator.go#L726
Listing PVCs used by prometheus via lables and selectors,  thanks.